### PR TITLE
Test all Known Coders to ensure they Serialize via URN

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Fix Coders to not encode VarIntCoders as VarLongCoders, and
VarLongCoders not as known coders.

Ensure that all known coders are tested.

Use getComponents rather than getCoderArguments for known coders.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @lukecwik

File: runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/CodersTest.java [hunk 1]
```diff
@@ -18,46 +18,88 @@
...
+          .add(IntervalWindowCoder.of())
+          .add(IterableCoder.of(ByteArrayCoder.of()))
+          .add(GlobalWindow.Coder.INSTANCE)
+          .add(
+              FullWindowedValueCoder.of(
+                  IterableCoder.of(VarLongCoder.of()), IntervalWindowCoder.of()))
+          .build();
+
+  @BeforeClass
+  public static void validateKnownCoders() {
```
Comment (lukecwik) [inline runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/CodersTest.java]: Use the Enclosed test runner and have two static nested classes, one containing this test and one containing the parameterized test. This way validateKnownCoders will not be incorrectly reported as a test error vs a test failure.
See:
http://stackoverflow.com/questions/8758294/test-cases-in-inner-classes-with-junit
or AvroIOTransformTest.java as well

**Comment by tgroh** [comment]
Done.

**Comment by lukecwik** [comment]
LGTM

**Pull request closed without merging**